eth/contracts/facets/DFSpaceshipFacet.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+// External contract imports
+import {DFWhitelistFacet} from "./DFWhitelistFacet.sol";
+import {DFTokenFacet} from "./DFTokenFacet.sol";
+
+// Library imports
+import {LibPermissions} from "../libraries/LibPermissions.sol";
+import {LibSpaceship} from "../libraries/LibSpaceship.sol";
+
+// Storage imports
+import {WithStorage, SnarkConstants, GameConstants} from "../libraries/LibStorage.sol";
+
+// Type imports
+import {Spaceship, SpaceshipType} from "../DFTypes.sol";
+
+contract DFSpaceshipFacet is WithStorage {
+    /**
+     * Modifiers
+     */
+    modifier onlyAdminOrCore() {
+        require(
+            msg.sender == gs().diamondAddress || msg.sender == LibPermissions.contractOwner(),
+            "Only the Core or Admin addresses can fiddle with artifacts."
+        );
+        _;
+    }
+
+    modifier onlyWhitelisted() {
+        require(
+            DFWhitelistFacet(address(this)).isWhitelisted(msg.sender) ||
+                msg.sender == LibPermissions.contractOwner(),
+            "Player is not whitelisted"
+        );
+        _;
+    }
+    /**
+     * Events
+     */
+    event ArtifactFound(address player, uint256 artifactId, uint256 loc);
```

Comment:
```suggestion
    event SpaceshipFound(address player, uint256 artifactId, uint256 loc);
```
I think we should separate spaceship events

Reply:
Ok, let me change the event